docs/src/main/asciidoc/_high-availability/cloud/Kubernetes.adoc

```diff
@@ -0,0 +1,230 @@
+<4> Configures preserving cookies as additional options for handling cookies.
+<5> The name of the cookie that contains the affinity information.
+<6> Allow backend servers to use a Set-Cookie HTTP header to emit their own affinity cookie value, meaning the backend servers have knowledge of which cookie value should route to which server.
+<7> Instructs the proxy to not modify/rewrite existing affinity cookie, only insert if a cookie is not set (e.g. if the backed server does not support storing affinity in a separate cookie)
+<8> Instructs the proxy what names to use as backed service names by respecting the `backend-server-naming` setting.
+
+== Verifying Session Affinity
+
+Verifying session affinity has proven to be notoriously challenging for users.
+This is primarily because of the fact, that even if the affinity handling is not set up correctly,
+the observer will not observe any data loss in a distributed HA application.
+This is simply because if the request for a given HTTP session is routed to a cluster node that does not have the session data locally,
+it will remotely fetch the session data from another node.
+While this does not affect functionality per se, this has implications for performance, response time, concurrent session access, etc.
+
+The following is a simple guide how to verify correctly functioning session affinity.
+
+First, configure the Undertow web server to attach information about which cluster node actually processed the request.
+This can be done by creating a `filter` that adds a header (called `JBoss-Node-Name` in this example) with a value of the `jboss.node.name` expression.
+This value is equal to the Kubernetes pod ID.
+
+[source,cli,title=undertow-filter.cli]
+----
+/subsystem=undertow/configuration=filter/response-header=node-name-header:add(header-name="JBoss-Node-Name",header-value=${jboss.node.name})
+/subsystem=undertow/server=default-server/host=default-host/filter-ref=node-name-header:add()
+----
+
+Secondly, after deploying any distributable application, query any URL that creates an HTTP session.
+
+[source,cli,title=undertow-filter.cli]
+----
+$ curl http://example.com:8080/clusterbench/session --verbose --insecure --cookie-jar cookies.txt --cookie cookies.txt
+*   Trying 127.0.0.1:8080...
+* Connected to localhost (127.0.0.1) port 8080 (#0)
+> GET /clusterbench/session HTTP/1.1
+> Host: localhost:8080
+> User-Agent: curl/8.1.2
+> Accept: */*
+>
+< HTTP/1.1 200 OK
+< Connection: keep-alive
+< JBoss-Node-Name: cb-85bf45d4c5-gnrxx # <1>
+< Set-Cookie: INGRESSCOOKIE=cb-85bf45d4c5-gnrxx; path=/ # <2>
+< Set-Cookie: JSESSIONID=EgxQStQ8zJ60lmDDO0VeY2H2OiLH3fdHRn2rqh5g; path=/clusterbench
+< X-JBoss-Node-Name: ribera
+< Content-Type: text/plain;charset=ISO-8859-1
+< Content-Length: 1
+< Date: Tue, 21 Nov 2023 14:58:48 GMT
+<
+* Connection #0 to host localhost left intact
+5
+----
+<1> This is the name of the node that actually processed the request that was added by the Undertow filter above.
```

Comment:
"This is the ~~name~~ _identity_ of the ~~node~~ _pod_ that actually processed the request ~~that was added by the Undertow filter above~~."

Reply:
Fixed.